test/FunctionalTests/Tests/DataServices/UnitTests/ClientCSharpUnitTests/DataWebClientCSharp/BuildingRequestEventFunctionalTests.cs

```diff
@@ -962,9 +962,14 @@ private string RetrieveLastSegmentOfRequestUri(ReceivingResponseEventArgs arg)
             {
                 return arg.Descriptor.State.ToString();
             }
+            else if (arg.Descriptor != null)
+            {
+                return arg.Descriptor.State.ToString();
+            }
             else
             {
-                return (arg.ResponseMessage as HttpWebResponseMessage).Response.ResponseUri.Segments.Last();
+                // this is the top-level batch request
+                return "$batch";

```

Comment:
Did `return "$batch" exist before though?

Reply:
For this test case, this statement is reached when it's a top-level batch request, in which case the `Response.ResponseUri.Segments.Last()` from the old code would return `$batch` as that's the last segment of a batch request URI. But since we don't have access to the `HttpWebResponse` now that we're using `HttpClienRequestMessage`, I've hardcoded `$batch` in this test.

A more robust solution would be to add a `RequestUri` property to the `ReceivingResponseEventArgs` and change the code that triggers the event to pass that information down. But I think that's a basically a new feature that probably would need its own discussion and PR.